source/material_model/rheology/elasticity.cc

```diff
@@ -269,98 +371,293 @@ namespace aspect
       template <int dim>
       void
       Elasticity<dim>::fill_reaction_outputs (const MaterialModel::MaterialModelInputs<dim> &in,
-                                              const std::vector<double> &average_elastic_shear_moduli,
+                                              const std::vector<double> &,
                                               MaterialModel::MaterialModelOutputs<dim> &out) const
       {
-        if (in.current_cell.state() == IteratorState::valid && this->get_timestep_number() > 0 && in.requests_property(MaterialProperties::reaction_terms))
+        if (in.current_cell.state() == IteratorState::valid
+            && in.requests_property(MaterialProperties::reaction_terms))
           {
-            // Get old (previous time step) velocity gradients
+            // Get velocity gradients of the current timestep $t+dtc$
+            // and the compositions from the previous timestep $t$,
+            // both at the requested location in in.position.
             std::vector<Point<dim>> quadrature_positions(in.n_evaluation_points());
-            for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
+            for (unsigned int i = 0; i < in.n_evaluation_points(); ++i)
               quadrature_positions[i] = this->get_mapping().transform_real_to_unit_cell(in.current_cell, in.position[i]);
 
+            // Get the current velocity gradients, which get
+            // updated in each nonlinear iteration.
+            // TODO Decide on whether to use W^t instead of W^(t+dtc)
+            // which is used now.
```

Comment:
What about this todo?

Reply:
I'll change it back to W^t, but I need to test that this is still the best option.

Reply:
Still need to test this, sorry :)